Resolves #10 (#30)
solid-software/help_ukraine_widget: README.md

# Help Ukraine Widget for Flutter

> This is a port of [Help Ukraine Widget](https://helpukrainewinwidget.org/) -

Reviewer: Style solid badge:

```suggestion
[![style: solid](https://img.shields.io/badge/style-solid-orange)](https://pub.dev/packages/solid_lints)

This is a port of [Help Ukraine Widget](https://helpukrainewinwidget.org/) -
```

# Help Ukraine Widget for Flutter
This is a port of [Help Ukraine Widget](https://helpukrainewinwidget.org/) -
this widget helps to let users of your app know how they can help Ukraine
from anywhere in the world.

> ### Some examples

Reviewer: ```suggestion
### Showcase
```

<details open><summary>First widget variation</summary>

![](gifs/widget1.gif)
</details>

<details><summary>Second widget variation</summary>
    
![](gifs/widget2.gif)
</details>

<details><summary>Third widget variation</summary>

![](gifs/widget3.gif)
</details>

<details><summary>Fourth widget variation</summary>

![](gifs/widget4.gif)
</details>

## Features
By installing this widget in your app, you will help to end the war in Ukraine.
It will provide links for:

* [donations](https://uahelp.monobank.ua/),
* sharing this widget
* and [other ways to support Ukraine](https://war.ukraine.ua/).

## Getting started
Add the package as a dependency to your __pubspec.yaml__:

``` yaml
dependencies:
    help_ukraine_widget: <latest version>
```

in your code:

``` dart
import 'package:help_ukraine_widget/help_ukraine_widget.dart';
```

## Usage

### Basic usage
Use `OverlayWidget` to float and align the widget above your app:

> ```

Reviewer: ```suggestion
``` dart
```

Author: done

OverlayWidget(
    // e.g. Alignment.bottomRight

>     alignment: alignment,

Reviewer: Simpler: 

```suggestion
    alignment: Alignment.bottomRight,
```

    overlayWidget: HorizontalHelpWidget(),
    child: MyApp(),
)
```

### Customization
You can also create custom variations of this widget:
``` dart
// we used that in our widget, but you do you 
final order = [HelpWidgetView.collapsed, HelpWidgetView.main, HelpWidgetView.options];

final _controller = TraverseController(order);

HelpWidget(
    controller: _controller;
    // view with links to resources that help Ukraine
    optionsView: ...

>     // transitional view between options and collapsed
>     mainView: ...

Reviewer: Not really logical to describe it as "transitional" -- this is the view that the users see first as they open the page.

```suggestion
    // The default first view that the users see.
    mainView: ...
```

# Help Ukraine Widget for Flutter
[![style: solid](https://img.shields.io/badge/style-solid-orange)](https://pub.dev/packages/solid_lints)

This is a port of [Help Ukraine Widget](https://helpukrainewinwidget.org/) -
this widget helps to let users of your app know how they can help Ukraine
from anywhere in the world.

### Showcase

<details open><summary>First widget variation</summary>

![](gifs/widget1.gif)
</details>

<details><summary>Second widget variation</summary>
    

> ![](gifs/widget2.gif)

Reviewer: <img width="521" alt="image" src="https://user-images.githubusercontent.com/40931732/188171975-b0bb7dae-57c5-418f-b21a-17a4d52039db.png">

Reviewer: Highly suggest using markdownlint

Author: added

final order = [HelpWidgetView.collapsed, HelpWidgetView.main, HelpWidgetView.options];

final _controller = TraverseController(order);

HelpWidget(
    controller: _controller;
    // view with links to resources that help Ukraine
    optionsView: ...
    // The default first view that the users see.
    mainView: ...
    // smallest view that doesn't obstruct user from
    // using the app, but still exists on the screen
    collapsedView: ...
    // axis of transition animation between views
    axis: ...
)
```

To make it easier we provide some classes:

#### Components

##### Buttons

`DetailsButton` - used in main view to go to options view.

`HelpOptionButton` - used in options view as a link to external resources.

`RoundedButton` - close button for `CardRounded`.

##### Other

`CardRounded` - base widget for every view.

`ChevronDown` - flexible chevron icon.

`FlagCard` - default collapsed view.

`FlagWidget` - flexible widget with Ukraine flag colors.

`HoverWrapper` - add responses for pointer events to your widget.

`XMark` - flexible cross icon.

### Controllers

`TraverseController` is used to track current view and direction of last transition.

### Helpers

> `defaultOptionsList` - list of default link buttons for external resources.

Reviewer: Let's probably scrap this.

High chance to get this out of sync with real implementations. Much more reliable to redirect to autogenerated API docs as soon as we're out on pub.dev.

# Help Ukraine Widget for Flutter
This is a port of [Help Ukraine Widget](https://helpukrainewinwidget.org/) -
this widget helps to let users of your app know how they can help Ukraine
from anywhere in the world.

### Some examples

<details open><summary>First widget variation</summary>

![](gifs/widget1.gif)
</details>

<details><summary>Second widget variation</summary>
    
![](gifs/widget2.gif)
</details>

<details><summary>Third widget variation</summary>

![](gifs/widget3.gif)
</details>

<details><summary>Fourth widget variation</summary>

![](gifs/widget4.gif)
</details>

## Features
By installing this widget in your app, you will help to end the war in Ukraine.
It will provide links for:

* [donations](https://uahelp.monobank.ua/),
* sharing this widget
* and [other ways to support Ukraine](https://war.ukraine.ua/).

## Getting started

> <h5 a><strong><code>pubspec.yaml</code></strong></h5>

Reviewer: HTML here is uncalled for, let's stick to plain Markdown wherever possible.
Besides, we could be a bit more verbose:
```suggestion
Add the package as a dependency to your __pubspec.yaml__:
```

``` yaml
dependencies:

>     help_ukraine_widget: ^1.0.0

Reviewer: We'll probably start with an unstable version. Besides -- we don't want to keep updating the version in the README each time we do a release.
```suggestion
    help_ukraine_widget: <latest version>
```

```

in your code:

``` dart
import 'package:help_ukraine_widget/help_ukraine_widget.dart';
```

## Usage

### Basic usage
`HorizontalHelpWidget()`

### Customization
You can add widget floating above your app like that:
```
OverlayWidget(
    alignment: alignment,
    overlayWidget: HorizontalHelpWidget(),
    child: MyApp(),
)

> ```

Reviewer: Let's be a bit more verbose here.

Basic usage involves baking it into user's layout -- let's make it explicit.

I'd also say thet using the OverlayWidget is a case of basic usage -- make sure to describe how to position the widget using it.

As for customization -- we should be explaining how the users could make their own layout, reusing widgets/controllers from our code.

* and [other ways to support Ukraine](https://war.ukraine.ua/).

## Getting started

Add the package as a dependency to your __pubspec.yaml__:

``` yaml
dependencies:
    help_ukraine_widget: <latest version>
```

in your code:

``` dart
import 'package:help_ukraine_widget/help_ukraine_widget.dart';
```

## Usage

### Basic usage

Use `OverlayWidget` to float and align the widget above your app:

``` dart
OverlayWidget(
    alignment: Alignment.bottomRight,
    overlayWidget: HorizontalHelpWidget(),
    child: MyApp(),
)
```

### Customization

You can also create custom variations of this widget:

``` dart
// we used that in our widget, but you do you 
final order = [HelpWidgetView.collapsed, HelpWidgetView.main, HelpWidgetView.options];

final _controller = TraverseController(order);

HelpWidget(
    controller: _controller;
    // view with links to resources that help Ukraine
    optionsView: ...
    // The default first view that the users see.
    mainView: ...
    // smallest view that doesn't obstruct user from
    // using the app, but still exists on the screen
    collapsedView: ...
    // axis of transition animation between views
    axis: ...
)
```

> To make it easier we provide some [classes](#).
> 

Reviewer: Let's not leave placeholder text in -- we'll add it when we get to it.
```suggestion
```
